### PR TITLE
Tendrils now spawn pits of lava instead of chasms

### PR DESCRIPTION
**Pull request by genessee596**
## About The Pull Request

Lavaland tendrils now melt into pits of lava instead of the instant-death chasms they did before.

## Why It's Good For The Game

I'm not going to sugarcoat it or try to dance around it; I've died to a chasm way too many times, and I'm mad enough about it to make a PR 'removing' it.

With that out of the way, let's actually sit down and discuss chasms - most notably, why they're still in the game and why I think they shouldn't be found naturally;

**Chasms currently exist because 'lavaland is supposed to be hard'.** No, seriously. That's the top reason I've seen when I've asked around about these. _Yes,_ lavaland is supposed to be hard. 
_But.._ there's a difference between 'fun, but challenging' and 'bullshit'. Getting killed by a huge demon because I got cocky or didn't high-tail it as soon as I saw it? That's fair. It's **your** fault you tried to kill the Demon King, and you deserve to be gibbed for it - or, you beat the monster, and get the reward. There's a reason why people take pride on being able to consistently beat fauna - it's fun, and it's not something everyone can do. 

**So** what do chasms bring to the table? Their main use is to force miners to be fast about grabbing the loot from a tendril, for no reason but added challenge. Yet - they are an instant-round remove. They're more effective than Bubblegum at making sure everyone who falls prey to them is booted from the rest of the round - at least with Bubblegum, you can salvage the victims' head for borging or something. With a chasm, you're done. Game over, man, game over! **All because you were too slow to move away from a chasm - or worse yet, didn't see the slightly-darker shade of black over the layers of ash.**

**How is that fun?** Simply put, it's not. You don't fall into a chasm and say, 'Ah, drat - oh, well, at least I learned how to better overcome this challenge in the future.' What happens is you fall into the chasm, say, 'What the fuck-- noooo!', and then have to watch everyone else have fun for a couple hours until the next round. _This is not fun. This is cruel - a risk with no real reward associated with it, as often miners die to the chasms after the fact than when finishing off a tendril._

**In conclusion,** I don't think chasms are fun and shouldn't be in the game. However, in order to keep a level of urgency to extracting a chest from a defeated tendril, the tendril will instead burst into a pool of lava and make getting the chest a pain in the ass to get if you aren't fast about it.
**TL:DR;** _I'm salty about dying to a chasm again_

## Changelog
:cl:
tweak: Tendrils now burst into pools of lava instead of bottomless pits.
/:cl:

### CODE REVIEW

**Comment by TemporalOroboros** [comment]
Can you also adjust it so that tendrils can't destroy other tendrils? I've lost too many because they spawned too close together.

**Comment by ATH1909** [comment]
https://www.youtube.com/watch?v=_J6-3l3hCm0

**Comment by ATH1909** [comment]
Just so you're aware, I'm pretty sure that tendril chests are lavaproof, so they won't actually get destroyed by the tendril's lava pool. It's probably fine, though.

**Comment by ATH1909** [comment]
Also, yes, we know that jaunters exist. However, non-golem lavaland ghost roles who DON'T beeline straight for the mining base and mining cyborgs both don't have access to jaunters, so they're still vulnerable to getting gg no re'd by chasms.

Speaking of jaunters, you should probably change their description and/or make them work on both lava pools AND chasms. Don't just flat-out remove them, though, as they still have their uses.

**Comment by genessee596** [comment]
> Also, yes, we know that jaunters exist. However, non-golem lavaland ghost roles who DON'T beeline straight for the mining base and mining cyborgs both don't have access to jaunters, so they're still vulnerable to getting gg no re'd by chasms.
> 
> Speaking of jaunters, you should probably change their description and/or make them work on both lava pools AND chasms. Don't just flat-out remove them, though, as they still have their uses.

Good point on the jaunters bit - if this does get merged, I'll expand their usefulness towards lava pools or make them automatically go off when you're critted or something, I dunno.

They still do have their uses, as you said, as they can be manually activated for an emergency teleport

**Comment by OmegaDarkPotato** [comment]
I've seen more than a couple friends of mine fall into a chasm due to a lagspike as well. I'm all in favor of making something a little more forgiving take the chasm's place.

**Comment by LemonInTheDark** [comment]
Come on travis, get yourself in gear

**Comment by Winterous** [comment]
I think a good first step would be to add a glow to chasms, so you can see them in the dark, and so they pop from the terrain a bit.

**Comment by TemporalOroboros** [comment]
To the sprite or to their lighting, because I think the chasm turfs already have the same glow as lava.

**Comment by wesoda25** [comment]
Glow or not, a turf that deletes you irreparably is the dumbest shit. This is a no brainer.

**Comment by Livrah** [comment]
The worst thing is to accidentally run into the chasm left by another miner

**Comment by genessee596** [comment]
> To the sprite or to their lighting, because I think the chasm turfs already have the same glow as lava.

They do not have any glow. If you lack night vision or have your gamma maxed, there's no way you'll see a chasm unless you're actively looking for one or you're directly next to one with a flashlight.

**Comment by TemporalOroboros** [comment]
https://github.com/tgstation/tgstation/blob/103bad6e9eaec79a77fdbb0dde942bf97d116e80/code/game/turfs/open/chasm.dm#L86-L92

I think they're supposed to emit light. If they're not that's probably a bug.

**Comment by genessee596** [comment]
> https://github.com/tgstation/tgstation/blob/103bad6e9eaec79a77fdbb0dde942bf97d116e80/code/game/turfs/open/chasm.dm#L86-L92
> 
> I think they're supposed to emit light. If they're not that's probably a bug.

Then yeah, they're bugged. I've never seen them emit light.

**Comment by ExcessiveUseOfCobblestone** [comment]
we can agree theres a difference between challenging and lol-nope difficulty and we can even agree chasms are a form of the less-enjoyable version of difficulty. 

Where we disagree is that i'm personally not keen with just removing it or lessening that overall difficulty because we dont like it as much.

Really though you should just fix the light bug first.

**Comment by CRITAWAKETS** [comment]
they do glow, its just that good luck looking for gray on gray and theres so much random shit on lavaland that glows anyway

**Comment by genessee596** [comment]
> Where we disagree is that i'm personally not keen with just removing it or lessening that overall difficulty because we dont like it as much.

Chasms are not fun, and short of a total rework on how chasms work, an instant-death turf will never be fun. I play Space Station 13 to have fun. Therefore, chasms should not exist.

That's my logic, anyhow.

**Comment by GuyonBroadway** [comment]
Tbh, the chasms really add nothing to gameplay, it's less making it harder and more of a lagcheck for miners. 

The only time you might lose a chest is if you lag or are too damaged to move quickly enough to snag it, and the latter occurs when you shoot a tendril just as a watcher spawns and hits you with a freezy-doodle. It adds nothing to skill based challange, just a noobgate or rng screw-over.

**Comment by Fikou** [comment]
> Where we disagree is that i'm personally not keen with just removing it or lessening that overall difficulty because we dont like it as much.

what difficulty? the fact that new miner players will be instadie because they dont know of the mechanic? the fact that if you dont pay attention you can fall into a barely visible hole and instadie?

**Comment by ExcessiveUseOfCobblestone** [comment]
being barely visible is a bug, which I suggested should be fixed.

It's a very opening/beginner-level difficulty in that you have to navigate around it, and where these chasms occur is the extent of that difficulty (maybe it was generated in how you get pass the lava river, etc.). It is an obstacle easily overcame by slowing down, yet people refuse to protect themselves against and allegedly dying to it left/right due to lag or whatever. I think the only time I have died to a chasm is twice: one where I happened to get snagged on a tentacle as I was pulling a chest out of a tendril and the other is when i left a tendril unkilled (hivelords) and when i was walking back someone had killed it, which may have would have been alleviated if the chasm light pr was implemented and working.

I'm **not** asking to rework chasms to make them work, i'm asking that this bad form of obstacle/difficulty be replaced, even if it's placed in a different aspect of lavaland, with what is believed to be a good form of obstacle/difficulty. 

Retaining the hyperlethality of Mining is important for me, especially since that is what made it so enticing to people when it was first added.

**Comment by genessee596** [comment]
> I'm **not** asking to rework chasms to make them work, i'm asking that this bad form of obstacle/difficulty be replaced, even if it's placed in a different aspect of lavaland, with what is believed to be a good form of obstacle/difficulty.

This _is_ "what is believed to be a good form of obstance/difficulty". Looking at all the other comments and reactions, people are totally alright with chasms being replaced with lava.

I honestly can't think of any other valid obstacle that could be reasonably - and fairly - used in this situation besides lava.

Anyways, I **won't** be changing this PR. I think it's perfectly fine as it is, and I think most everyone else agrees with me on this.

**Comment by wesoda25** [comment]
Deleting a person for stepping on the wrong tile isn’t hyper lethality, it’s retarded. Hyper lethality is stepping on the wrong tile, taking massive damage, and catching fire. There’s challenge, and then there is shit game design.

**Comment by GuyonBroadway** [comment]
I honestly don't think the "stuff happening when a tendril dies" is much of a challange anywho, it's just a glorified quicktime event that just punishes lag or those who got slowed just before killing the tendril. It has exactly 1 solution and its "don't be hurt or lagging." Which just isn't very engaging to play around.

**Comment by LemonInTheDark** [comment]
> 
> This _is_ "what is believed to be a good form of obstance/difficulty". Looking at all the other comments and reactions, people are totally alright with chasms being replaced with lava.
> 
> I honestly can't think of any other valid obstacle that could be reasonably - and fairly - used in this situation besides lava.
> 
> Anyways, I **won't** be changing this PR. I think it's perfectly fine as it is, and I think most everyone else agrees with me on this.

I don't intend to step on cobbys toes here, but I'd like to elaborate on something.

Mining is currently precariously balanced by the fact that miners die constantly in dumb ways.
If it weren't for this, we would have even more issues with the gamer gear we give them. This means that we need to at least upkeep this challenge so things don't get too bad. In light of this, I'd like to ask you to add something to balance out this removal, somewhere, could be in another pr even.

If you don't want to do this I will be closing this pr.

**Comment by Sikroth** [comment]
You say that "Mining is currently precariously balanced by the fact that miners die constantly in dumb ways" but I am inclined to think the kind of player that makes mining so troublesome to balance (aka the minmaxing powergamer getting all the sweet gamer gear asap) is most likely not dying to these chasms anyway.

I don't think making tendril somewhat less lethal if you happen to misstep is going to suddenly break anything. And keep in mind, if you're not prepared (extiniguisher, etc) stepping on a single lava tile is probably going to kill you anyway too. You'll just have a slight chance of maybe being brought back if someone's kind enough to get you.

**Comment by genessee596** [comment]
> > This _is_ "what is believed to be a good form of obstance/difficulty". Looking at all the other comments and reactions, people are totally alright with chasms being replaced with lava.
> > I honestly can't think of any other valid obstacle that could be reasonably - and fairly - used in this situation besides lava.
> > Anyways, I **won't** be changing this PR. I think it's perfectly fine as it is, and I think most everyone else agrees with me on this.
> 
> I don't intend to step on cobbys toes here, but I'd like to elaborate on something.
> 
> Mining is currently precariously balanced by the fact that miners die constantly in dumb ways.
> If it weren't for this, we would have even more issues with the gamer gear we give them. This means that we need to at least upkeep this challenge so things don't get too bad. In light of this, I'd like to ask you to add something to balance out this removal, somewhere, could be in another pr even.
> 
> If you don't want to do this I will be closing this pr.

You disgust me.

**Comment by wesoda25** [comment]
Bruh.

**Comment by ExcessiveUseOfCobblestone** [comment]
You can disagree with someone but if you are going to result to rude comments about the person/people who disagree with you don't expect your stay to be long or your current leave to be short.
